src/main/java/com/google/sps/utils/QuizTimingPropertiesUtils.java

```diff
@@ -65,22 +64,19 @@ public Object getTimestampProperty(String entity, DatastoreService datastore) {
         }
         if(pq.asList(FetchOptions.Builder.withLimit(1)).size() > 0) {
             Entity fetched_item = pq.asList(FetchOptions.Builder.withLimit(1)).get(0);
-            return fetched_item.getProperty("quiz_timestamp");
+            return (Long) fetched_item.getProperty("quiz_timestamp");
         } 
         log.severe("Zero Items Quered");
         return null;
     }
  
     //This function checks if the user has taken the quiz yet by comparing their timestamp with the quiz's timestamp
     public Boolean userTookQuiz(String usersQuizTime, String currentQuizTime) {
-        if(usersQuizTime.compareTo(currentQuizTime) > 0) {
-            return true;
-        } 
-        return false;
+        return (usersQuizTime.compareTo(currentQuizTime) > 0);
     }
 
     //This function checks to see if the quiz is outdated
-     public Boolean newDayNewQuiz(Object current_quiz_time) {
+     public Boolean isQuizOutdated(Object current_quiz_time) {
         String quiz_date;
         try {
             quiz_date = DateFormat.getDateInstance().format(current_quiz_time);
```

Comment:
The behavior of this code depends on the configured time zone of the computer the code runs on. That can lead to problems when your code is running in the cloud. I recommend either specifying a time zone when converting the `Long` to a date, or else comparing the timestamps without converting them to dates.

Reply:
Okay I did the latter suggestion, let me know if that is what you meant